app/models/interview.rb

```diff
@@ -0,0 +1,48 @@
+class Interview < ApplicationRecord
+  validates :round, :participant1, :participant2, :starts_at, :ends_at, presence: true
+  validate :participants_should_be_different
+  validate :date_cannot_be_in_the_past
+  validate :no_overlap
+
+  has_attached_file :resume
+  validates_attachment_presence :resume
+  validates_attachment :resume, presence: true, content_type: { content_type: "application/pdf" }
+
+  after_create :remind
+  after_update :remind_update_mail
+  after_destroy :remind_delete_mail
+
+  private
+    def remind
+      ReminderMailer.remind_mail(self).deliver_later(wait_until: (self.starts_at - DateTime.now - 30.minutes).seconds.from_now)
+    end
+    def remind_update_mail
+      ReminderMailer.remind_update(self).deliver_now
+      ReminderMailer.remind_mail(self).deliver_later(wait_until: (self.starts_at - DateTime.now - 30.minutes).seconds.from_now)
```

Comment:
How are we ensuring the previous scheduled mail is prevented once the interview is rescheduled?